e2e2/internal/metric/metric.go

```diff
@@ -0,0 +1,192 @@
+	}
+
+	// Set appropriate headers for the request
+	req.Header.Set("Content-Type", "application/x-protobuf")
+	req.Header.Set("Content-Encoding", "snappy")
+	req.Header.Set("X-Prometheus-Remote-Write-Version", "0.1.0")
+
+	// Create a SigV4 signer
+	signer := v4.NewSigner()
+	payloadHash, err := utils.CreatePayloadHash(req)
+	if err != nil {
+		return fmt.Errorf("failed to create payload hash: %w", err)
+	}
+
+	//extract region from remoteWriteURL
+	region, err := extractRegionFromURL(m.AmpRemoteWriteURL)
+	if err != nil {
+		return fmt.Errorf("failed to extract region from URL: %w", err)
+	}
+
+	// Sign the HTTP request using the credentials from the assumed role
+	err = signer.SignHTTP(context.TODO(), creds, req, payloadHash, "aps", region, time.Now())
+	if err != nil {
+		return fmt.Errorf("failed to sign the request: %w", err)
+	}
+
+	// Send the HTTP request
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to send HTTP request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	// Read and print the response body
+	res, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("failed to read response body: %w", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("received non-OK response code: %d, response body: %s", resp.StatusCode, string(res))
+	}
+
+	return nil
+}
+
+// convertMetrics converts a slice of MetricFamily to snappy compressed Protobuf data
+// Reference: https://github.com/prometheus/prometheus/blob/1c624c58ca934f618be737b4995e22051f5724c1/prompb/remote.pb.go#L64
+// https://github.com/aws-observability/observability-best-practices/blob/main/sandbox/CWMetricStreamExporter/lambda/main.go#L264
+func convertMetrics(mfs []*dto.MetricFamily) (*bytes.Reader, error) {
+	var timeSeries []prompb.TimeSeries
+
+	// Loop over each MetricFamily and convert it into Prometheus TimeSeries format
+	for _, mf := range mfs {
+		for _, metric := range mf.Metric {
+			labels := []prompb.Label{
+				{Name: "__name__", Value: mf.GetName()},
+			}
+
+			// Add ConstLabels to TimeSeries
+			for _, v := range metric.GetLabel() {
+				labels = append(labels, prompb.Label{
+					Name:  v.GetName(),
+					Value: v.GetValue(),
+				})
+			}
+			ts := prompb.TimeSeries{
+				Labels: labels,
+				Samples: []prompb.Sample{
+					{
+						Value:     metric.GetGauge().GetValue(),
+						Timestamp: time.Now().UnixMilli(),
+					},
+				},
+			}
+			timeSeries = append(timeSeries, ts)
+		}
+	}
+
+	// Build WriteRequest with TimeSeries data
+	writeRequest := &prompb.WriteRequest{
+		Timeseries: timeSeries,
+	}
+
+	// Marshal WriteRequest to Protobuf
+	data, err := proto.Marshal(writeRequest)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal write request: %w", err)
+	}
+
+	// Snappy compress the Protobuf data
+	compressed := snappy.Encode(nil, data)
+	body := bytes.NewReader(compressed)
+
+	return body, nil
+}
+
+func extractRegionFromURL(remoteWriteURL string) (string, error) {
```

Comment:
Could you just require the user to pass in the ampRegion? this regex will need to be updated for other+future partitions, etc